source/base/data_out_base.cc

```diff
@@ -613,42 +632,44 @@ namespace
     else if (patch.reference_cell_type == ReferenceCell::Type::Tri &&
              patch.data.n_cols() == 3)
       {
-        vtk_cell_id[0] = 5;
+        vtk_cell_id[0] = VTK_TRIANGLE;
         vtk_cell_id[1] = 1;
       }
     else if (patch.reference_cell_type == ReferenceCell::Type::Tri &&
              patch.data.n_cols() == 6)
       {
-        vtk_cell_id[0] = 22;
+        vtk_cell_id[0] = VTK_QUADRATIC_TRIANGLE;
         vtk_cell_id[1] = 1;
       }
     else if (patch.reference_cell_type == ReferenceCell::Type::Tet &&
              patch.data.n_cols() == 4)
       {
-        vtk_cell_id[0] = 10;
+        vtk_cell_id[0] = VTK_TETRA;
         vtk_cell_id[1] = 1;
       }
     else if (patch.reference_cell_type == ReferenceCell::Type::Tet &&
              patch.data.n_cols() == 10)
       {
-        vtk_cell_id[0] = 24;
+        vtk_cell_id[0] = VTK_QUADRATIC_TETRA;
         vtk_cell_id[1] = 1;
       }
     else if (patch.reference_cell_type == ReferenceCell::Type::Wedge &&
              patch.data.n_cols() == 6)
       {
-        vtk_cell_id[0] = 13;
+        vtk_cell_id[0] = VTK_WEDGE;
         vtk_cell_id[1] = 1;
       }
     else if (patch.reference_cell_type == ReferenceCell::Type::Pyramid &&
              patch.data.n_cols() == 5)
       {
-        vtk_cell_id[0] = 14;
+        vtk_cell_id[0] = VTK_PYRAMID;
         vtk_cell_id[1] = 1;
       }
     else if (patch.reference_cell_type == ReferenceCell::get_hypercube(dim))
       {
-        vtk_cell_id[0] = vtk_cell_type[dim];
+        const std::array<unsigned int, 3> cell_type_by_dim{
+          {VTK_LINE, VTK_QUAD, VTK_HEXAHEDRON}};
+        vtk_cell_id[0] = cell_type_by_dim[dim - 1];
```

Comment:
You either write:
```suggestion
        vtk_cell_id[0] = cell_type_by_dim[std::max(0, dim - 1)];
```
or you introduce `VTK_VERTEX`.